projects/core/src/anonymous-consents/store/effects/anonymous-consents.effect.ts

```diff
@@ -171,9 +171,9 @@ export class AnonymousConsentsEffects {
           const actions: UserActions.GiveUserConsent[] = [];
           for (const template of templates) {
             if (
-              (!template.currentConsent ||
-                !template.currentConsent.consentGivenDate ||
-                template.currentConsent.consentWithdrawnDate) &&
+              this.userConsentService.isConsentWithdrawn(
+                template.currentConsent
```

Comment:
this logic is removed (!template.currentConsent.consentGivenDate), is it fine?

Reply:
Yes, it's essentially the same logic, but moved to the facade